### PR TITLE
Issue #5513: Create xdoc page to explain the new Xpath-based suppression model

### PR DESCRIPTION
**Pull request by timurt**
https://github.com/checkstyle/checkstyle/issues/5513

### CODE REVIEW

File: src/xdocs/config_filters.xml [hunk 1]
```diff
@@ -860,6 +855,60 @@ public class UserService {
 &lt;suppress-xpath checks=&quot;.*&quot; query=&quot;/CLASS_DEF[@text='TestClass']
           //METHOD_DEF[@text='testMethod']//VARIABLE_DEF[@text='testVariable']&quot;/&gt;
         </source>
+        <p>
+          In the following sample, violations for <code>LeftCurly</code> check will be suppressed
+          for classes with name <i>Main</i> or for methods with name <i>calculate</i>.
+        </p>
+        <source>
+&lt;suppress-xpath checks=&quot;LeftCurly&quot; query=&quot;/CLASS_DEF[@text='Main']/OBJBLOCK/LCURLY
+          | //METHOD_DEF[@text='calculate']/SLIST&quot;/&gt;
```
Comment (romani) [inline src/xdocs/config_filters.xml]: users will be lazy to make xpath very specific , can we make xpath like:
`/CLASS_DEF[@text='Main']/* | //METHOD_DEF[@text='calculate']/*` 
to  not bother user to know what token Check actually report an violation.

File: src/xdocs/config_filters.xml [hunk 2]
```diff
@@ -860,6 +855,60 @@ public class UserService {
...
+  public void changeAge() {
+    age = 24; //violation will be suppressed
+  }
+}
+        </source>
+        <p>
+          Suppress <code>IllegalThrows</code> violations only for methods with name
+          <i>throwsMethod</i> and only for <code>RuntimeException</code> exceptions.
+          Double colon is used for axis iterations. In the following example <code>ancestor</code>
+          axis is used to iterate all ancestor nodes of the current node with type
```
Comment (romani) [inline src/xdocs/config_filters.xml]: please provide some links where Xpath is explained (all axises, ... ) and all supported query features of it. Do not expect that users are know Xpath very well. Even I do not know it :).

File: src/xdocs/config_filters.xml [hunk 3]
```diff
@@ -860,6 +855,60 @@ public class UserService {
...
+  }
+
+  public void sampleMethod() throws RuntimeException { // will throw violation here
+  }
+}
+        </source>
+        <p>
+          The following sample demonstrates how to suppress all violations for method itself and
+          all descendants. <code>descendant-or-self</code> axis iterates through current node and
+          all children nodes at any level. Keyword <code>node()</code> selects node elements.
```
Comment (romani) [inline src/xdocs/config_filters.xml]: please make link to supported keywords.

File: src/xdocs/config_filters.xml (hunk 2 above)
Comment (timurt) [reply src/xdocs/config_filters.xml]: What do you think about [w3schools](https://www.w3schools.com/xml/xpath_axes.asp) link?

Comment (romani) [reply src/xdocs/config_filters.xml]: looks good, lets start at least from it.

Comment (timurt) [reply src/xdocs/config_filters.xml]: done

File: src/xdocs/config_filters.xml (hunk 3 above)
Comment (timurt) [reply src/xdocs/config_filters.xml]: done

File: src/xdocs/config_filters.xml (hunk 1 above)
Comment (timurt) [reply src/xdocs/config_filters.xml]: done

File: src/xdocs/config_filters.xml [hunk 4]
```diff
@@ -860,6 +855,63 @@ public class UserService {
...
+        <p>
+          In the following sample, violations for <code>LeftCurly</code> check will be suppressed
+          for classes with name <i>Main</i> or for methods with name <i>calculate</i>.
+        </p>
+        <source>
+&lt;suppress-xpath checks=&quot;LeftCurly&quot; query=&quot;/CLASS_DEF[@text='Main']//*
+          | //METHOD_DEF[@text='calculate']/*&quot;/&gt;
+        </source>
+        <p>
+          The following example demonstrates how to suppress <code>RequireThis</code> errors for
```
[the remark this answers is not in the document]
Comment (timurt) [reply src/xdocs/config_filters.xml]: Ok, should I change just my line or all "errors" occurrences in this file?
See [this](https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/config_filters.xml#L540), 
[this](https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/config_filters.xml#L543) and [this](https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/config_filters.xml#L545)

**Pull request merged**